metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -1297,3 +1301,64 @@ def visit_semi_additive_join_node(self, node: SemiAdditiveJoinNode) -> SqlDataSe
                 order_bys=(),
             ),
         )
+
+    def visit_join_to_time_spine_node(self, node: JoinToTimeSpineNode[SourceDataSetT]) -> SqlDataSet:  # noqa: D
+        parent_data_set = node.parent_node.accept(self)
+        parent_alias = self._next_unique_table_alias()
+
+        # Build time spine dataset
+        metric_time_dimension_instance: Optional[TimeDimensionInstance] = None
+        for instance in parent_data_set.metric_time_dimension_instances:
+            if len(instance.spec.identifier_links) == 0:
+                metric_time_dimension_instance = instance
+                break
+        assert (
+            metric_time_dimension_instance
+        ), "Can't query offset metric without a time dimension. Validations should have prevented this."
+        metric_time_dimension_column_name = self.column_association_resolver.resolve_time_dimension_spec(
+            metric_time_dimension_instance.spec
+        ).column_name
+        time_spine_alias = self._next_unique_table_alias()
+        time_spine_dataset = _make_time_spine_data_set(
+            metric_time_dimension_instance=metric_time_dimension_instance,
+            metric_time_dimension_column_name=metric_time_dimension_column_name,
+            time_spine_source=self._time_spine_source,
+            time_spine_table_alias=time_spine_alias,
+            time_range_constraint=node.time_range_constraint,
+        )
+
+        # Build join expression
+        left_expr: SqlExpressionNode = SqlColumnReferenceExpression(
+            col_ref=SqlColumnReference(table_alias=time_spine_alias, column_name=metric_time_dimension_column_name)
+        )
+        if node.offset_window:
+            left_expr = SqlTimeDeltaExpression(
+                arg=left_expr, count=node.offset_window.count, granularity=node.offset_window.granularity
+            )
+        elif node.offset_to_grain_to_date:
+            left_expr = SqlDateTruncExpression(time_granularity=node.offset_to_grain_to_date, arg=left_expr)
```

Comment:
See comment on the SQL - this does the date trunc on the time spine value, not the metric time value. So the EQUALS below filters all of the intervening values. I believe we need the date_trunc on the other side of the join comparison as well, so we probably need to build up a right_expr here.

Reply:
We're good on this, as discussed in slack!